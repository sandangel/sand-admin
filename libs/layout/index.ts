export { LayoutModule } from './src/layout.module';
export { LayoutState, createSetSidenav, createCloseSidenav, createOpenSidenav } from './src/+state';
