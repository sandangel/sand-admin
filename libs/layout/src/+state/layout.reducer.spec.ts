import { layoutReducer, layoutInitialState, Layout } from './layout.reducer';
import { createSetSidenav, createOpenSidenav } from './layout.actions.helpers';

describe('layoutReducer', () => {
  it('should work', () => {
    const state: Layout = layoutInitialState;
    const action1 = createSetSidenav(true);
    const actual = layoutReducer(state, action1);
    expect(actual).toEqual({ sidenavOpened: true });
  });

  it('should return init state by default', () => {
    const action = createOpenSidenav();
    const actual = layoutReducer(layoutInitialState, action);
    expect(actual).toEqual(layoutInitialState);
  });

  it('should have initial state if not provided', () => {
    const action = createOpenSidenav();
    expect(layoutReducer(undefined, action)).toEqual(layoutInitialState);
  });
});
