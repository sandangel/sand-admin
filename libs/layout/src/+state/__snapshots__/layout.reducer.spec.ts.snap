// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layoutReducer should match snapshots 1`] = `
Object {
  "sidenavOpened": true,
}
`;
